packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-comment-box.tsx

```diff
@@ -93,14 +115,33 @@ export const PullRequestCommentBox = ({
   const [textSelection, setTextSelection] = useState({ start: 0, end: 0 })
   const dropZoneRef = useRef<HTMLDivElement>(null)
 
+  const [localPrincipalsMentionMap, setLocalPrincipalsMentionMap] = useState<PrincipalsMentionMap>({})
+
+  const {
+    principalsMentionMap,
+    setPrincipalsMentionMap
+  }: {
+    principalsMentionMap: PrincipalsMentionMap
+    setPrincipalsMentionMap: React.Dispatch<React.SetStateAction<PrincipalsMentionMap>>
+  } = useMemo(() => {
+    return {
+      principalsMentionMap: parentPrincipalsMentionMap ?? localPrincipalsMentionMap,
+      setPrincipalsMentionMap: parentSetPrincipalsMentionMap ?? setLocalPrincipalsMentionMap
+    }
+  }, [parentPrincipalsMentionMap, localPrincipalsMentionMap, parentSetPrincipalsMentionMap])
+
   const handleTabChange = (tab: typeof TABS_KEYS.WRITE | typeof TABS_KEYS.PREVIEW) => {
     setActiveTab(tab)
   }
 
   const handleSaveComment = () => {
     if (comment.trim()) {
-      onSaveComment(comment)
-      setComment('') // Clear the comment box after saving
+      const formattedComment = replaceMentionEmailWithId(comment, principalsMentionMap)
+      onSaveComment(formattedComment)
+
+      if (!preserveCommentOnSave) {
```

Comment:
This component is being used for description as well. In that scenario, we need to preserve the state for edit purpose. Hence `preserveCommentOnSave` has been added.